exchanges/binance/binance_wrapper.go

```diff
@@ -1911,3 +1923,537 @@ func (b *Binance) GetServerTime(ctx context.Context, ai asset.Item) (time.Time,
+func (b *Binance) GetFuturesPositionOrders(ctx context.Context, req *order.PositionsRequest) ([]order.PositionResponse, error) {
+	if req == nil {
+		return nil, fmt.Errorf("%w PositionSummaryRequest", common.ErrNilPointer)
+	}
+	if len(req.Pairs) == 0 {
+		return nil, currency.ErrCurrencyPairsEmpty
+	}
+	req.StartDate = req.StartDate.Truncate(time.Hour)
+	req.EndDate = req.EndDate.Truncate(time.Hour)
+	if time.Since(req.StartDate) > b.Features.Supports.MaximumOrderHistory {
+		if req.RespectOrderHistoryLimits {
+			req.StartDate = time.Now().Add(-b.Features.Supports.MaximumOrderHistory).Round(time.Hour)
+		} else {
+			return nil, fmt.Errorf("%w max lookup %v", order.ErrOrderHistoryTooLarge, time.Now().Add(-b.Features.Supports.MaximumOrderHistory))
+		}
+	}
+	if req.EndDate.IsZero() {
+		req.EndDate = time.Now()
+	}
+
+	var resp []order.PositionResponse
+	sd := req.StartDate
+	switch req.Asset {
+	case asset.USDTMarginedFutures:
+		var orderLimit = 1000
+		for x := range req.Pairs {
+			fPair, err := b.FormatExchangeCurrency(req.Pairs[x], req.Asset)
+			if err != nil {
+				return nil, err
+			}
+			result, err := b.UPositionsInfoV2(ctx, fPair)
+			if err != nil {
+				return nil, err
+			}
+			for y := range result {
+				currencyPosition := order.PositionResponse{
+					Asset: req.Asset,
+					Pair:  req.Pairs[x],
+				}
+				for {
+					var orders []UFuturesOrderData
+					orders, err = b.UAllAccountOrders(ctx, fPair, 0, int64(orderLimit), sd, time.Time{})
+					if err != nil {
+						return nil, err
+					}
+					for i := range orders {
+						if orders[i].Time.After(req.EndDate) {
+							continue
+						}
+						orderVars := compatibleOrderVars(orders[i].Side, orders[i].Status, orders[i].OrderType)
+						currencyPosition.Orders = append(currencyPosition.Orders, order.Detail{
+							ReduceOnly:           orders[i].ClosePosition,
+							Price:                orders[i].Price,
+							Amount:               orders[i].ExecutedQty,
+							TriggerPrice:         orders[i].ActivatePrice,
+							AverageExecutedPrice: orders[i].AvgPrice,
+							ExecutedAmount:       orders[i].ExecutedQty,
+							RemainingAmount:      orders[i].OrigQty - orders[i].ExecutedQty,
+							CostAsset:            req.Pairs[x].Quote,
+							Leverage:             result[y].Leverage,
+							Exchange:             b.Name,
+							OrderID:              strconv.FormatInt(orders[i].OrderID, 10),
+							ClientOrderID:        orders[i].ClientOrderID,
+							Type:                 orderVars.OrderType,
+							Side:                 orderVars.Side,
+							Status:               orderVars.Status,
+							AssetType:            asset.USDTMarginedFutures,
+							Date:                 orders[i].Time,
+							LastUpdated:          orders[i].UpdateTime,
+							Pair:                 req.Pairs[x],
+							MarginType:           result[y].MarginType,
+						})
+					}
+					if len(orders) < orderLimit {
+						break
+					}
```

Comment:
Just from first look this looks out of place like it won't continue anymore orders if there are say 5 / 10. This seems like it will process the first one then break and don't worry about the other 4. I could be wrong, time for another coffee.

Reply:
🎉 yeah this needs adjusting. Have another coffee to celebrate